src/csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs

```diff
@@ -218,6 +219,8 @@ private ISocketWrapper GetConnection()
                     methodName,
                     args);
 
+                _socketSemaphore.Wait();
```

Comment:
Can this just be moved to the top of this `try` block ?  Then there would be no need to introduce the `bool` flag.

Reply:
It could. I was under the impression that there was a reason you wanted to grab the connection later. I agree that moving it up top would simplify the code and was my initial idea, but I assumed the connection was established later on purpose. I don't see any other fields mutated before GetConnection being used by GetConnection so it seems like a safe operation. If you agree let me know and I'll go ahead and change.

Reply:
I think it should be safe to go ahead and move the wait to the top.

Reply:
Pushed a fix. Please review.